external_parser/event_processors/joined_event.h

```diff
@@ -310,6 +310,103 @@ struct ccb_joined_event : public typed_joined_event {
   }
 };
 
+struct DecisionServiceInteractionCats {
+  std::string eventId;
+  std::string timestamp;
+  float action;
+  float pdf_value;
+  float probabilityOfDrop = 0.f;
+  bool skipLearn{false};
+};
+
+struct ca_joined_event : public typed_joined_event {
+  DecisionServiceInteractionCats interaction_data;
+  float reward = 0.0f;
+  float original_reward = 0.0f;
+
+  ~ca_joined_event() = default;
+
+  bool is_skip_learn() const override { return interaction_data.skipLearn; }
+
+  void set_skip_learn(bool sl) override { interaction_data.skipLearn = sl; }
+
+  void set_apprentice_reward() override {
+    if (!std::isnan(interaction_data.action)) {
+      // TODO: default apprenticeReward should come from config
+      // setting to default reward matches current behavior for now
+      reward = original_reward;
+    }
+  }
+
+  void fill_in_label(v_array<example *> &examples) const override {
+
+    if (std::isnan(interaction_data.action)) {
+      VW::io::logger::log_warn("missing action for event [{}]",
+                               interaction_data.eventId);
+      return;
+    }
+
+    if (std::isnan(interaction_data.pdf_value)) {
+      VW::io::logger::log_warn("missing pdf_value for event [{}]",
+                               interaction_data.eventId);
+      return;
+    }
+
+    example *ex = examples[0];
```

Comment:
probably worth checking here `if examples.size() > 0` and if not log and return (otherwise we will throw badly)

Reply:
Just added a check